src/components/Todo.tsx

```diff
@@ -0,0 +1,82 @@
+import classNames from 'classnames';
+import { remove } from '../api/todos';
+import { Todo } from '../types/Todo';
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+interface Props {
+  completed: boolean,
+  title: string;
+  id: number;
+  setError: (value: string) => void,
+  setTodos: (value: Todo[]) => void,
+  todos: Todo[],
+  setSelectedTodoId: (value: number) => void,
+  selectedTodoId: number,
+}
+
+export const TodoInfo: React.FC<Props> = ({
+  completed,
+  title,
+  id,
+  setTodos,
+  setError,
+  todos,
+  setSelectedTodoId,
+  selectedTodoId,
+}) => {
+  const handlerClick = (removeId: number) => {
+    setSelectedTodoId(removeId);
+    const fetchData = async () => {
+      try {
+        await remove(removeId);
+
+        setTodos([...todos.filter(todoo => todoo.id !== removeId)]);
```

Comment:
If the new state is computed using the previous state, you can pass a function to setTodos (and not just here)